src/components/media-loader.js

```diff
@@ -6,6 +6,8 @@ const fetchContentType = async url => fetch(url, { method: "HEAD" }).then(r => r
 AFRAME.registerComponent("media-loader", {
   schema: {
     src: { type: "string" },
+    token: { type: "string" },
+    contentType: { type: "string" },
```

Comment:
Hmm, feels like we shouldn't need contentType, part of media-loaders job is figuring that out so it feels a bit out of place to tell it. It should get this back from `resolveMedia` anyway.

Reply:
When `url` matches `uploads-dev.reticulum.io/uploads/`, `resolveMedia` only returns `original` and `raw` (which is the same value). It does not return `meta` with `expected_content_type`.
 
If alternatively I skip `resolveMedia` and call `fetchContentType` (with the authorization header added to the request), the `OPTIONS` returns a `204` and the `HEAD` returns a `401`. Neither of these have a `Content-Type` header.

Running into some difficulty tracing thru this in reticulum. I don't understand that code very well.

Reply:
If I could pass the `File` to `media-loader`, I could have it use `upload` instead of `resolveMedia`, which is basically what we want. I think we only consider using `resolveMedia`/`fetchContentType` because that's the existing code path inside `media-loader`. Since `media-loader` is an `aframe` `component` whose state we want to sync via `networked-aframe`, we eventually do need to set its `src` and `token` attributes. However, when we set the `src` to the `url` of an uploaded `File`, we don't need to call `resolveMedia` or `fetchContentType`, since the `upload` request returned the expected content type and the url we want to network. 
I'm not sure what the right way to think about this is. On Friday I was trying to think thru whether `media-loader` needs to be a component that does work, since the work we need to do changes depending on what media we want to load, but I didn't think of a clean way to avoid doing work. We want the loading cube to exist the whole time we're uploading or downloading media, which means we want it to be on the same entity that the eventually loaded image/video/model will be attached to.

Reply:
the reticulum media API should return expected content type in the meta section of the response when you upload a file -- is that not working? was hard to tell from these comments.

Reply:
OK I think I understand now. I will work on a reticulum patch to properly respond to HEAD from the uploads controller, which will include the content type.

Reply:
actually it looks to me like this already works. my guess is you are not passing the authorization header when you make the HEAD request?

Reply:
working on a PR

Reply:
https://github.com/mozilla/hubs/pull/505